File: go/models/post.go
package models

import "time"

type Post struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Content     string    `json:"content"`
	UserID      uint      `json:"userId"`                        // Foreign key
	User        User      `gorm:"foreignKey:UserID" json:"user"` // Relationship
	CreatedAt   time.Time `json:"createdAt"`
}

File: rust/src/model/post.rs
use async_graphql::{Result, SimpleObject};
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize, SimpleObject)]
#[sea_orm(table_name = "post")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = true)]
    pub id: i32,
    pub title: String,
    pub description: String,
    pub content: String,
    pub user_id: i32,
}

// pub type Post = Model;
// pub type PostEntity = Entity;

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
  #[sea_orm(
    belongs_to = "super::user::Entity",
    from = "Column::UserId",
    to = "super::user::Column::Id"
  )]
  User,
}

impl Related<super::user::Entity> for Entity {
  fn to() -> RelationDef {
      Relation::User.def()
  }
}



impl ActiveModelBehavior for ActiveModel {}
